packages/@aws-cdk/aws-appmesh/lib/tls-certificate.ts

```diff
@@ -121,3 +138,28 @@ class FileTlsCertificate extends TlsCertificate {
     };
   }
 }
```

Comment:
I think I came up with a way to make the "ACM can't be used in mutual TLS" condition compile-time safe.

Have an abstract class that extends `TlsCertificate`:

```ts
export abstract class MutualTlsCertificate extends TlsCertificate {
  // TypeScript uses structural typing, so we need any property different from TlsCertificate
  private readonly differentiator = false;
}
```

Then, change the types returned from `TlsCertificate.sds()` and `TlsCertificate.file()` to be `MutualTlsCertificate`, and require that type in the new `mutualTlsCertificate` property that you're adding in this PR to `TlsClientPolicy`.

Presto, now passing `TlsCertficiate.acm()` to `mutualTlsCertificate` will not compile!

The same basic trick can also be used for `TlsValidation`, and `mutualTlsValidation`.

Reply:
This is brilliant idea.
I was not able to come up with a way to add compile-error but this does do a work!